Add a custom protocol over SSH for TCP and UDP communication

lunrsh opened the pull request:
# Summary
This introduces a new backend (& custom protcol) called sshpy.

Commit:
chore: Gets basic server working for SSHPy.

Commit:
fix: Fixes backend creation.

Co-authored-by: dev <hi@dessa.dev>

Commit:
chore: Adds username field in user creation API route.

Commit:
feature: Adds more preliminary support for sshpy.

Co-authored-by: dev <hi@dessa.dev>

Commit:
fix: Fixes code to pass eslint.

Commit:
feature: Gets basic TCP working in SSHpy.

It's not very stable lmao

Co-authored-by: dev <hi@dessa.dev>

Commit:
fix: Also add disconnecting support.

Commit:
fix: Fixes most crashing and freezing issues.

Co-authored-by: dev <hi@dessa.dev>

Commit:
feature: Implement UDP support.

Commit:
fix: Fixes missing break statement.

Commit:
chore: Formats code.

Commit:
chore: Reorder imports.

Commit:
feature: Adds support for forward rule closing.

greysoh commented:
Before I merge this, I need to implement these things (afaik):
* I'm using 32 bit numbers for no reason. I should probably switch this to 16 bits, but maybe keep this for the client IDs
* I probably should write documentation on the protocol and the code, since it isn't really off the shelf, and the code is ~1.3k lines so far in total

devawaves commented:
This PR will fix issue #9.

lunrsh committed:
chore: Clarify description.

Commit:
fix: Makes protocol use 16 bit integers instead of 32 bit integers.

Excluding TCP client IDs, of course. Additionally, this isn't fully stable right now.

Co-authored-by: dev <hi@dessa.dev>
